packages/object/src/index.ts

```diff
@@ -18,6 +19,24 @@ export interface CRO {
 	mergeCallback: (operations: Operation[]) => void;
 }
 
+export class CROState {
+	cro: CRO;
+	linearizedOperations: Operation[];
+	// the past includes all vertices that the vertex depends on and the vertex itself
+	// root has in-degree 0
+	past: Set<Hash>;
+
+	constructor(
+		cro: CRO,
+		linearizedOperations: Operation[] = [],
+		past: Set<Hash> = new Set(),
```

Comment:
```suggestion
		linearizedOperations: Operation[],
		past: Set<Hash>,
```

there's no case that the constructor should be passed without this inputs